### ISSUE TITLE
Plugin support

### ISSUE THREAD

**Issue by NothingAnimated** [opentoonz/opentoonz#4019]
In other opensource software like krita and blender they have support for custom made plugins. They usually add some type of function like in krita there is a plugin to be able to export spritesheets. In opentoonz I think adding plugin support would be a really  
useful feature as this will allow people to make plugins for opentoonz without having to fork it out waiting for it to be added as a feature, they can just make a plugin that people can download and install.

**Comment by NothingAnimated** [comment]
And im not specifically talking about fx plugins(if they are made differently)

**Comment by NothingAnimated** [comment]
unless this feature is already in opentoonz.

**Comment by DarrenTAnims** [comment]
There is a plugin feature in OpenToonz, which is used for creating new effects. I don't know if you can use it for anything else.

**Comment by absozero** [comment]
I like the idea for custom made plugins. What I also like, is custom brushes and maybe importing/borrowing the brush engines from krita. They are highly advanced and are insane when painting in krita. For me, it is a real pain in the butt to have to switch between krita and openttoz for drawing+animation.

**Comment by NothingAnimated** [comment]
@DarrenTAnims Yeah I know about the fx plugins and those are amazing and i think expanding that to being able to add functionality to other parts of opentoonz will greatly improve the software.

**Comment by NothingAnimated** [comment]
@absozero they have the my paint brush sets and im pretty sure nobel frugel studio (on youtube) has a tutorial on adding in custom brushes.

**Comment by DarrenTAnims** [comment]
Yeah. Other plugin types would be good, but aren't likely to be added until more developers join the project, unfortunately.

**Comment by RodneyBaker** [comment]
It appears that the framework for plugins in Krita is python so perhaps this request might start with yet another request... support for python in Opentoonz.

REF:  https://docs.krita.org/en/user_manual/python_scripting/krita_python_plugin_howto.html

**Comment by RodneyBaker** [comment]
I would think that python support in Opentoonz might logically be accomplished via Qtpy.
Qtpy 6.0 was released back in December 2020.

Here is a link for that:  https://www.qt.io/blog/qt-for-python-6-released

**Comment by RodneyBaker** [comment]
For compiling the python into binary executables it would seem Nuitka is well worth investigating:

https://nuitka.net/

**Comment by NothingAnimated** [comment]
@RodneyBaker I thought the pluggin api would have to be C++ because opentoonz was C++. but yeah if possible python would help with that making it more accesible.

**Comment by RodneyBaker** [comment]
Conducting a review of open issues/requests.
While important, this request for 'additional plugin support' is overly broad in nature with no actionable objectives to implement.

Recommend deeper review of Opentoonz current and projected plugin system be initiated during v1.8 development cycle to provide developers with a path to get from here (current plugin system) to there (long term plugin architecture).  Additionally, suggestions to aid users in suitable workflow while such development is ongoing should be provided so users need not wait until that development is fully implmented.  (ex:  How to add other programs into the production pipeline along with Opentoonz to meet the immediate needs that a more robust internal plugin system would provide).

**Comment by RodneyBaker** [comment]
Closing this as too generic to implement but keep these ideas at the ready for additional discussion.  We likely won't get much additional plugin supoort into version 1. 8 but I'm confident by version 2.0 we should have a pretty solid foundation.
